test/front-end.test.js

```diff
@@ -0,0 +1,83 @@
+const Application = require('spectron').Application;
+const assert = require('assert');
+
+describe('Front-end tests', function () {
+    this.timeout(20000); // needed for Travis builds
+
+    beforeEach(() => {
+        this.app = new Application({
```

Comment:
I think using `this` in a test case is a bit obscure, Could you please create a `const` variable `app` in the describe and use that?

Reply:
Could you show me an example of what that would look like?

Reply:
```
describe("...",function(){
let app; //Sorry, it should be let, not const
beforeEach(()=>{
    app=new Application
    ....
})
```

This way, you explicitly know what variables are using, and the internals of the suite this won't collide